students/ikempcode33/lesson 5/database.py

```diff
@@ -0,0 +1,99 @@
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        self.host = host
+        self.port = port
+        self.connection = None
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+# Read in CSV data
+def import_data(directory_name, product_file, customer_file, rentals_file):
+    """Takes in csv files, counts customers, products and rentals and errors"""
+    mongo = MongoDBConnection()
+    # set to zero
+    customer_errors = 0
+    product_errors = 0
+    rental_errors = 0
+    customer_count = 0
+    product_count = 0
+    rental_count = 0
+    with mongo:
+        db = mongo.connection.media
+        customer = db["customer"]
+        product = db["product"]
+        rental = db["rental"]
+        customer.drop
+        product.drop
+        rental.drop
+
+        try:
+            with open(os.path.join(directory_name, customer_file)) as csv_file:
+                cust_reader = csv.reader(csv_file, delimiter=',')
+                for row in cust_reader:
+                    customer_count += 1
+                    customer_info = {'customer_id': row[0], 'name': row[1], 'address': row[2],
+                                     'phone': row[3], 'email': row[4]}
+                    customer.insert_one(customer_info)
+        except FileNotFoundError:
+            customer_errors += 1
+        try:
+            with open(os.path.join(directory_name, product_file)) as csv_file:
+                prod_read = csv.reader(csv_file, delimiter=',')
+                for row in prod_read:
+                    product_count += 1
+                    product_info = {'product_id': row[0], 'description': row[1], 'product_type': row[2],
+                                    'quantity': row[3]}
+                    product.insert_one(product_info)
+        except FileNotFoundError:
+            product_errors += 1
+        try: 
+            with open(os.path.join(directory_name, rentals_file)) as csv_file:
+                rent_read = csv.reader(csv_file, delimiter=',')
+                for row in rent_read:
+                    rental_count += 1
+                    info = {'customer_id': row[0], 'product_id': row[1]}
+                    rental.insert_one(info)
+        except FileNotFoundError:
+            rental_errors += 1
+    return ((customer_count, product_count, rental_count),
+            (customer_errors, product_errors, rental_errors))
+
+def show_available_products():
+    """Returns a dictionary of available products"""
+    mongo = MongoDBConnection()
+    with mongo:
+        db = mongo.connection.media
+        products = db['products']
+        available_products = {}  #Dictionary output
+        # Iterate through products for dict
+        for item in products.find():
+            available_products[item['product_id']] = {'description': item['description'],
```

Comment:
Note that in your database, you really don't have a _product_type_ or _quantity_ fields. Instead, you have this:

```
> db.products.find()
{ "_id" : ObjectId("5ec867e8ab332d42405e42b4"), "product_id" : "2001", "description" : "table_lamp", "type" : "electronics", "number_available" : "4" }
```
So, _product_type_ is just _type_ and _quantity_ is _number_available_. So, first step will be to correct that. At least on my side, I get other errors on the comparison, but let's tackle one thing at a time.